store/store_action.go

```diff
@@ -342,14 +378,39 @@ func (s *Store) snapAction(ctx context.Context, currentSnaps []*CurrentSnap, act
 		actionJSONs[i] = aJSON
 	}
 
+	// assertions
+	i := len(actions)
```

Comment:
This could be moved down to the scope of `if len(toResolve) > 0 {`. 
But perhaps more readable would be to have single counter for both snaps and assertions loops and not to have to declare or re-set it here.